pkg/upgrade/drain_manager.go

```diff
@@ -83,7 +83,7 @@ func (m *DrainManagerImpl) ScheduleNodesDrain(ctx context.Context, drainConfig *
 		GracePeriodSeconds:  -1,
 		Timeout:             time.Duration(drainSpec.TimeoutSecond) * time.Second,
 		PodSelector:         drainSpec.PodSelector,
-		OnPodDeletedOrEvicted: func(pod *corev1.Pod, usingEviction bool) {
+		OnPodDeletionOrEvictionFinished: func(pod *corev1.Pod, usingEviction bool, _ error) {
```

Comment:
Looks like the idea is to evaluate `error` field to determine if the eviction finished or failed. We should change the below logic accordingly. See e.g. [here](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/drain/drain.go#L165).

Reply:
Agreed. Thanks for pointing this out!

Reply:
Please check now